modules/nw-ne-dynamic-config-manager.adoc

```diff
@@ -0,0 +1,84 @@
+// Module included in the following assemblies:
+//
+// * networking/routes/route-configuration.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ne-dynamic-config-manager_{context}"]
+= Enabling the dynamic configuration manager
+
+Some template routers, like HAProxy, use a dynamic configuration manager to propagate route and endpoint changes through a dynamic API. This approach allows the underlying router to adapt to changes efficiently and ensures that the system can scale and adjust as needed. The HAProxy manager uses this dynamic API to modify the operational state of HAProxy backends. Endpoint changes, such as scaling and node evictions, are managed by provisioning each backend with a pool of dynamic servers. The size of this dynamic servers pool is controlled by the max-dynamic-servers option or the ROUTER_MAX_DYNAMIC_SERVERS environment variable.
```

Comment:
```suggestion
The router uses a dynamic configuration manager to propagate endpoint changes using dynamic HAPRoxy API. This approach allows the router to adapt to changes efficiently reducing the number of reloads needed and hence the memory footprint. The dynamic configuration manager uses the HAProxy dynamic API to modify the operational state of HAProxy backends. Endpoint changes, such as scaling up and down, are managed by provisioning each backend with a pool of dynamic servers.
```

- We don't support other router templates by the HAProxy so we can omit the details about different templates.
- The configuration of the number of dynamic server is unsupported so we should not mention it.